internal/core/src/index/Index.h

```diff
@@ -14,19 +14,25 @@
 #include <memory>
 #include <knowhere/index/Index.h>
 #include <knowhere/common/Dataset.h>
-#include <knowhere/index/structured_index_simple/StructuredIndex.h>
+#include "index/OperatorType.h"
+#include <boost/dynamic_bitset.hpp>
 
 namespace milvus::scalar {
 using Index = knowhere::Index;
 using IndexPtr = std::unique_ptr<Index>;
 using BinarySet = knowhere::BinarySet;
 using Config = knowhere::Config;
 using DatasetPtr = knowhere::DatasetPtr;
-using OperatorType = knowhere::scalar::OperatorType;
+using TargetBitmap = boost::dynamic_bitset<>;
+using TargetBitmapPtr = std::unique_ptr<TargetBitmap>;
 
 class IndexBase : public Index {
+ public:
     virtual void
-    Build(const DatasetPtr& dataset) = 0;
+    BuildWithDs(const DatasetPtr& dataset) = 0;
```

Comment:
BuildWithIDs ?

Reply:
Build with dataset (DS).

Reply:
Better use BuildWithDataset

Reply:
`Build(dataset)` sometimes will conflict with the `Build(n, values)`. So maybe `BuildWithDataset` is better.